From b54d2c7d1f3d3c1666e40cfce603dcf1b4caf8c4 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 23:53:41 -0400
Subject: [PATCH] Add IllegalPacketEvent

Fired for invalid data from players that usually represents hacking

Import Patch from EMC
---
 .../paperdragon/event/IllegalPacketEvent.java      | 41 ++++++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 src/main/java/com/domnian/paperdragon/event/IllegalPacketEvent.java

diff --git a/src/main/java/com/domnian/paperdragon/event/IllegalPacketEvent.java b/src/main/java/com/domnian/paperdragon/event/IllegalPacketEvent.java
new file mode 100644
index 0000000..4788270
--- /dev/null
+++ b/src/main/java/com/domnian/paperdragon/event/IllegalPacketEvent.java
@@ -0,0 +1,41 @@
+package com.domnian.paperdragon.event;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class IllegalPacketEvent extends Event {
+    private final Player player;
+    private final String type;
+    private final Exception ex;
+
+    public IllegalPacketEvent(Player player, String type, Exception e) {
+        this.player = player;
+        this.type = type;
+        ex = e;
+    }
+
+    public Player getPlayer() {
+        return player;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public Exception getException() {
+        return ex;
+    }
+
+    private static final HandlerList handlers = new HandlerList();
+
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+}
+
+
-- 
2.5.0

